aten/src/ATen/core/jit_type.h

```diff
@@ -1384,9 +1385,42 @@ struct ClassType;
 using ClassTypePtr = std::shared_ptr<ClassType>;
 using ::torch::jit::script::CompilationUnit;
 using ::torch::jit::Function;
+struct SerializableType;
+using SerializableTypePtr = std::shared_ptr<SerializableType>;
+
+struct CAFFE2_API SerializableType : public Type {
```

Comment:
This should not inherit from Type. We don't want to muck up the type system with implementation details like this. Maybe make it a mixin